### PR TITLE
Adds headpats

### PR DESCRIPTION
**Pull request by monster860**
# General Documentation

### Intent of your Pull Request

When you click someone with help intent with their head selected, it gives them a "pats the head" message, instead of the other options, as well as causing a tail wag. 

### Why is this change good for the game?

Allow more options for using help intent. It also gives more options for things to do when there's down time and nothing is going on. More RP options in general.

# Wiki Documentation

### Briefly describe your PR and the impacts of it, in layman's terms. 

Using help intent with head will cause a headpat and a tail wag

### What should players be aware of when it comes to the changes your PR is implementing?
It causes tail wag

### What general grouping does this PR fall under? 
General RP stuff

### Are there any aspects of the PR that you would like us not to mention on the Wiki?
no

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 

# Changelog

:cl:  
add: Adds headpats, which is done by using help intent targeting the head
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -269,6 +269,17 @@
...
+	else if(check_zone(M.zone_selected) == BODY_ZONE_HEAD)
+		var/datum/species/S
+		if(ishuman(src))
+			S = dna.species
+
+		M.visible_message("<span class='notice'>[M] gives [src] a pat on the head to make [p_them()] feel better!</span>", \
+					"<span class='notice'>You give [src] a pat on the head to make [p_them()] feel better!</span>", target = src,
+					target_message = "<span class='notice'>[M] gives you a pat on the head to make you feel better!</span>")
+		if(S?.can_wag_tail(src) && !dna.species.is_wagging_tail())
+			emote("wag")
```
Comment (Hopekz) [inline code/modules/mob/living/carbon/carbon_defense.dm:281]: Epic

**Comment by ynot01** [comment]
does not compile

**Comment by maxion12345** [comment]
if this gets merged I'm making a PR to bring dabbing and tposing back.

**Comment by Mqiib** [comment]
Please no

**Comment by ChesterTheCheesy** [comment]
-forces target into wagging it's tail
bruh that's literally ERP

**Comment by Szyszkrzyneczka** [comment]
Burn the heretic, Kill the mutant, Purge the unclean,

the fact that it has a (wag) action makes me vomit, just add pinball monster

**Comment by Hopekz** [comment]
> if this gets merged I'm making a PR to bring dabbing and tposing back.

Tposing is lame dab good

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -269,6 +269,17 @@
 	else if(check_zone(M.zone_selected) == BODY_ZONE_L_ARM || check_zone(M.zone_selected) == BODY_ZONE_R_ARM) //Headpats are too extreme, we have to pat shoulders on yogs
 		M.visible_message("<span class='notice'>[M] gives [src] a pat on the shoulder to make [p_them()] feel better!</span>", \
 					"<span class='notice'>You give [src] a pat on the shoulder to make [p_them()] feel better!</span>")
+	
+	else if(check_zone(M.zone_selected) == BODY_ZONE_HEAD)
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm:273]:
```suggestion
	
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -269,6 +269,17 @@
...
+		var/datum/species/S
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm:274]:
```suggestion
		
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 4]
```diff
@@ -269,6 +269,17 @@
...
+		if(ishuman(src))
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm:275]:
```suggestion
		
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 5]
```diff
@@ -269,6 +269,17 @@
...
+			S = dna.species
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm:276]:
```suggestion
			
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 6]
```diff
@@ -269,6 +269,17 @@
...
+
+		M.visible_message("<span class='notice'>[M] gives [src] a pat on the head to make [p_them()] feel better!</span>", \
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm:278]:
```suggestion
		
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 7]
```diff
@@ -269,6 +269,17 @@
...
+					"<span class='notice'>You give [src] a pat on the head to make [p_them()] feel better!</span>", target = src,
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
					
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 8]
```diff
@@ -269,6 +269,17 @@
...
+					target_message = "<span class='notice'>[M] gives you a pat on the head to make you feel better!</span>")
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm]:
```suggestion
					
```

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 9]
```diff
@@ -269,6 +269,17 @@
...
+		if(S?.can_wag_tail(src) && !dna.species.is_wagging_tail())
```
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm:280]:
```suggestion
		
```

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 1 above)
Comment (Bibby0110) [inline code/modules/mob/living/carbon/carbon_defense.dm:281]:
```suggestion
			
```

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
looks good just a few tiny changes

**Comment by ChesterTheCheesy** [comment]
You can headpat a headless person, immersion destroyed.

**Comment by N3D6** [comment]
yeah no can we please not, we already have shoulderpats that nobody fucking uses

**Comment by alexkar598** [comment]
I want this so I can be condescending in game, head patting people.


There there, I fixed you up, dont get hurt again, it's bad for you, ok? *pats head*

**Comment by jcatjmeow** [comment]
Doesn't pass linter.
We are under a feature freeze.
Bad changelog.
Missing rationale.
Did not fill out the pr template.
Abusing maint powers to add the bug label and remove feature label.

**Comment by monster860** [comment]
I have fixed the changelog and made it follow the PR template and hopefully it should compile, although I did have to remove the targeted message, something I could re-add if theres enough demand

**Comment by Mqiib** [comment]
If this somehow gets merged I'm making a pr to remove the tail wag

**Comment by N3D6** [comment]
i'd dislike this pr a lot less if it didn't make you tailwag but this just seems like a shitpost if i'm being honest

**Comment by SomeguyManperson** [comment]
https://user-images.githubusercontent.com/24857008/106217650-b047ef80-61a3-11eb-953a-8aa84c69a11e.mp4

**Comment by Jarod1200** [comment]
TheOS has once again saved yogstation from imminent destruction

**Comment by monster860** [comment]
image

**Comment by SomeguyManperson** [comment]
the "damn good reason" is this is a meme pr stop trying to stir shit
to elaborate we don't need headpats for the same reason we don't need dabbing, tposing or other memey shit that adds basically nothing to the game; what people CAN do will impact what they WILL do and stupid catgirl anime emotes is the opposite of what we want people to do

**Comment by swissloaf** [comment]
No

**Review by jcatjmeow — CHANGES_REQUESTED** [review]
Tested it personally. Works. This will be merged over my dead body (also feature freeze) but it's fine as a feature sooooooooooo yeah

Request re-review after the feature freeze or somethin idk

**Comment by JamieD1** [comment]
Line ------

Dont cross it. @yogstation13/maintainers
